gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -136,6 +142,33 @@ public Single<PasswordPolicy> update(ReferenceType referenceType, String referen
                         .type(EventType.PASSWORD_POLICY_UPDATED).throwable(error)));
     }
 
+    @Override
+    public Completable delete(ReferenceType referenceType, String referenceId, String policyId, User principal) {
+        log.debug("Delete password policy with id '{}' for {} {}", policyId, referenceType, referenceId);
+        return passwordPolicyRepository.findByReferenceAndId(referenceType, referenceId, policyId)
+                .flatMap(policy ->
```

Comment:
Could you change 3 flatMap which returns the same object to one? 

findWithPassowrdPolicy -> updatePasswordPolicy -> delete -> create event  -> Maybe.just(policy)